### ISSUE TITLE
[Suggestion] Landing pad logic output

### ISSUE THREAD

**Issue by Alx-179** [tsunamayo/Starship-EVO#2632]
### Short Description of the suggestion:
It would very useful if you could get a logic output from a landing pad/shipyard depending on
 if landing is requested, a ship is docked or the pad is free.
### Example:
Maybe you could connect a query gate and then chose the output conditions from there.
These could be:
Landing requested:[True/False]
Ship docked: [True/False]
Landing pad free: [True/False]

This is probably a little much for me to ask for but it would enable some really cool mechanisms.
Nonetheless the game is really making great progress and I love it already. Keep up the great work Tsunamayo. 👍

**Comment by Garrett-C** [comment]
You can technically tell if the pad is free by using the volume sensor thing I would imagine. I can’t see how you would get three different types of logic signals from the current system. 

That said I think free and landed work as the true false for one statement (which as mentioned you should be able to get from a volume sensor) meaning you could set the landing pad up with only the ability to output a signal about docking requests.

**Comment by Alx-179** [comment]
"I can’t see how you would get three different types of logic signals from the current system."
I think you misunderstood me a little. These [true/false] options are not the possible outputs but options you could set in the menu of the query block. For example if
Landing requested:True
Ship docked: False
Landing pad free: True
is selected the query gate would output a signal when landing is requested or the pad is free but not when a ship is docked.

**Comment by Garrett-C** [comment]
Yeah I mean most of the functionality can be got from the volume sensor no? All you really need is a way to tell if docking is requested. I also still feel like ship docked and landing pad free are 2 sides of the same coin.

**Comment by Alx-179** [comment]
Yes, you are right. Making the shipyard just output when docking is requested would have the same functionality as the thing I suggested but the volumetric sensor would need a "block only" mode so it doesn't get activated by players (or NPCs) walking into it.

**Comment by Garrett-C** [comment]
A player only mode would indeed be a good addition. I don’t think it’s essential to achieve the functionality but it would definitely be a good addition.

**Comment by tsunamayo** [comment]
Yes this is planned for the big query refactor update. Cheers

**Comment by Kaiser-Indrasil** [comment]
Speaking of Query Gate refactor, I would also very much welcome the ability to feed the Query Gate the status of the Lightcruise - whether the ship is currently in it or not. That would work very well with the turret-hiding mechanism I have in mind.
